optimum/neuron/modeling_decoder.py

```diff
@@ -0,0 +1,208 @@
+            model_path = model_dir.name
+            # We are in the middle of an export: the checkpoint is in the temporary model directory
+            checkpoint_path = model_path
+            # There are no compiled artifacts yet
+            compiled_path = None
+        else:
+            model_path = model_dir
+            # The model has already been exported, the checkpoint is in a subdirectory
+            checkpoint_path = os.path.join(model_path, cls.CHECKPOINT_DIR)
+            # So are the compiled artifacts
+            compiled_path = os.path.join(model_path, cls.COMPILED_DIR)
+        return model_path, checkpoint_path, compiled_path
+
+    @classmethod
+    def _from_pretrained(
+        cls, model_id: Union[str, Path, TemporaryDirectory], config: "PretrainedConfig", **kwargs
+    ) -> "NeuronDecoderModel":
+        # Verify we are actually trying to load a neuron model
+        neuron_config = getattr(config, "neuron", None)
+        if neuron_config is None:
+            raise ValueError(
+                "The specified directory does not contain a neuron model. "
+                "Please convert your model to neuron format by passing export=True."
+            )
+
+        # Evaluate the configuration passed during export
+        task = neuron_config["task"]
+        neuron_kwargs = neuron_config["neuron_kwargs"]
+
+        exporter = get_exporter(config, task)
+
+        model_path, checkpoint_path, compiled_path = cls._get_neuron_paths(model_id)
+
+        neuronx_model = exporter.neuronx_class.from_pretrained(checkpoint_path, **neuron_kwargs)
+
+        if compiled_path is not None:
+            # Specify the path where compiled artifacts are stored before conversion
+            neuronx_model._load_compiled_artifacts(compiled_path)
+
+        # Compile the Neuron model (if present compiled artifacts will be reloaded instead of compiled)
+        os.environ["NEURON_CC_FLAGS"] = "--model-type=transformer-inference"
+        neuronx_model.to_neuron()
+
+        # Try to reload the generation config (if any)
+        generation_config = None
+        try:
+            generation_config = GenerationConfig.from_pretrained(model_path)
+        except OSError:
+            logger.info("Generation config file not found, using a generation config created from the model config.")
+
+        return cls(neuronx_model, config, model_id, generation_config)
+
+    def forward(self, *args, **kwargs):
+        raise NotImplementedError()
+
+    def _save_pretrained(self, save_directory: Union[str, Path]):
+        _, src_chkpt_path, src_compiled_path = self._get_neuron_paths(self.model_path)
+        _, dst_chkpt_path, dst_compiled_path = self._get_neuron_paths(save_directory)
+
+        shutil.copytree(src_chkpt_path, dst_chkpt_path)
+
+        if src_compiled_path is None:
+            # The compiled model has never been serialized: do it now
+            self.model._save_compiled_artifacts(dst_compiled_path)
```

Comment:
Why don't we serialize the compiled model anyway under `from_transformers` after compilation? Then here we have fewer cases to consider.

Reply:
Because in from_transformers we don't have the path to where it should be saved. But I agree that everybody will do a save_pretrained after a call to from_pretrained(export=True).